--- ceph-12.2.3/cmake/modules/BuildBoost.cmake.orig	2018-02-21 14:13:32.527899631 -0500
+++ ceph-12.2.3/cmake/modules/BuildBoost.cmake	2018-02-21 14:14:04.796899631 -0500
@@ -75,7 +75,7 @@
   set(install_command
     ${b2} install)
   set(boost_root_dir "${CMAKE_BINARY_DIR}/boost")
-  if(EXISTS "${PROJECT_SOURCE_DIR}/src/boost/libs/config/include/boost/config.hpp")
+  if(EXISTS "${PROJECT_SOURCE_DIR}/src/boost/boost/config.hpp")
     message(STATUS "boost already in src")
     set(source_dir
       SOURCE_DIR "${PROJECT_SOURCE_DIR}/src/boost")
