--- ceph-15.2.8/src/CMakeLists.txt.orig	2021-01-03 12:47:00.275274985 -0500
+++ ceph-15.2.8/src/CMakeLists.txt	2021-01-03 12:51:17.957185433 -0500
@@ -23,10 +23,15 @@
 set(datadir ${CEPH_INSTALL_DATADIR})
 set(prefix ${CMAKE_INSTALL_PREFIX})
 
-add_definitions("-DHAVE_CONFIG_H -D__CEPH__ -D_REENTRANT -D_THREAD_SAFE -D__STDC_FORMAT_MACROS")
-add_definitions("-D_FILE_OFFSET_BITS=64")
+add_definitions(-DHAVE_CONFIG_H
+  -D__CEPH__
+  -D_REENTRANT
+  -D_THREAD_SAFE
+  -D__STDC_FORMAT_MACROS
+  -D_FILE_OFFSET_BITS=64
+  -DBOOST_ASIO_DISABLE_THREAD_KEYWORD_EXTENSION)
 if(LINUX)
-  add_definitions("-D_GNU_SOURCE")
+  add_definitions(-D_GNU_SOURCE)
 endif()
 
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
