--- ceph-12.2.11/src/rgw/CMakeLists.txt.orig	2019-01-30 10:51:26.000000000 -0500
+++ ceph-12.2.11/src/rgw/CMakeLists.txt	2019-02-11 06:02:06.748608099 -0500
@@ -177,9 +177,7 @@
 
 add_library(radosgw_a STATIC ${radosgw_srcs}
   $<TARGET_OBJECTS:civetweb_common_objs>)
-if (WITH_RADOSGW_BEAST_FRONTEND AND WITH_RADOSGW_BEAST_OPENSSL)
-  target_link_libraries(radosgw_a rgw_a ${SSL_LIBRARIES})
-endif()
+target_link_libraries(radosgw_a rgw_a ${SSL_LIBRARIES})
 
 add_executable(radosgw rgw_main.cc)
 target_link_libraries(radosgw radosgw_a librados
@@ -195,10 +193,6 @@
   cls_version cls_replica_log cls_user)
 install(TARGETS radosgw DESTINATION bin)
 
-if (WITH_RADOSGW_BEAST_FRONTEND)
-  target_link_libraries(radosgw_a ${OPENSSL_LIBRARIES})
-endif()
-
 set(radosgw_admin_srcs
   rgw_admin.cc
   rgw_orphan.cc)
