--- ceph-16.2.7/src/ceph_mgr.cc.orig	2022-01-21 09:15:21.977384085 -0500
+++ ceph-16.2.7/src/ceph_mgr.cc	2022-01-21 10:32:17.586105697 -0500
@@ -14,6 +14,7 @@
  *
  */
 
+#include <limits.h>
 #include <Python.h>
 
 #include <pthread.h>
--- ceph-16.2.7/src/mgr/ActivePyModule.h.orig	2022-01-21 10:25:39.355471107 -0500
+++ ceph-16.2.7/src/mgr/ActivePyModule.h	2022-01-21 10:26:04.233010937 -0500
@@ -16,7 +16,8 @@
 #pragma once
 
 // Python.h comes first because otherwise it clobbers ceph's assert
-#include "Python.h"
+#include <limits.h>
+#include <Python.h>
 
 #include "common/cmdparse.h"
 #include "common/LogEntry.h"
--- ceph-16.2.7/src/mgr/BaseMgrModule.cc.orig	2022-01-21 09:16:58.807597191 -0500
+++ ceph-16.2.7/src/mgr/BaseMgrModule.cc	2022-01-21 10:18:27.335462311 -0500
@@ -18,7 +18,8 @@
  * available as methods on that object.
  */
 
-#include "Python.h"
+#include <limits.h>
+#include <Python.h>
 
 #include "Mgr.h"
 
--- ceph-16.2.7/src/mgr/BaseMgrStandbyModule.h.orig	2022-01-21 10:27:53.142996531 -0500
+++ ceph-16.2.7/src/mgr/BaseMgrStandbyModule.h	2022-01-21 10:16:57.530123467 -0500
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <limits.h>
 #include <Python.h>
 
 extern PyTypeObject BaseMgrStandbyModuleType;
--- ceph-16.2.7/src/mgr/DaemonServer.h.orig	2022-01-21 09:17:40.606825834 -0500
+++ ceph-16.2.7/src/mgr/DaemonServer.h	2022-01-21 10:18:45.760121505 -0500
@@ -14,6 +14,7 @@
 #ifndef DAEMON_SERVER_H_
 #define DAEMON_SERVER_H_
 
+#include <limits.h>
 #include "PyModuleRegistry.h"
 
 #include <set>
--- ceph-16.2.7/src/mgr/Gil.cc.orig	2022-01-21 09:14:35.203247250 -0500
+++ ceph-16.2.7/src/mgr/Gil.cc	2022-01-21 10:18:53.226983388 -0500
@@ -13,7 +13,8 @@
  */
 
 
-#include "Python.h"
+#include <limits.h>
+#include <Python.h>
 
 #include "common/debug.h"
 
--- ceph-16.2.7/src/mgr/Mgr.cc.orig	2022-01-21 09:18:30.161911483 -0500
+++ ceph-16.2.7/src/mgr/Mgr.cc	2022-01-21 10:17:19.178723027 -0500
@@ -11,6 +11,7 @@
  * Foundation.  See file COPYING.
  */
 
+#include <limits.h>
 #include <Python.h>
 
 #include "osdc/Objecter.h"
--- ceph-16.2.7/src/mgr/Mgr.h.orig	2022-01-21 09:16:27.837168716 -0500
+++ ceph-16.2.7/src/mgr/Mgr.h	2022-01-21 10:17:41.816304290 -0500
@@ -15,6 +15,7 @@
 #define CEPH_MGR_H_
 
 // Python.h comes first because otherwise it clobbers ceph's assert
+#include <limits.h>
 #include <Python.h>
 
 #include "mds/FSMap.h"
--- ceph-16.2.7/src/mgr/MgrStandby.cc.orig	2022-01-21 10:20:42.502962076 -0500
+++ ceph-16.2.7/src/mgr/MgrStandby.cc	2022-01-21 10:20:59.642645037 -0500
@@ -11,6 +11,7 @@
  * Foundation.  See file COPYING.
  */
 
+#include <limits.h>
 #include <Python.h>
 
 #include "common/errno.h"
--- ceph-16.2.7/src/mgr/PyFormatter.h.orig	2022-01-21 10:24:55.335285362 -0500
+++ ceph-16.2.7/src/mgr/PyFormatter.h	2022-01-21 10:25:06.758074070 -0500
@@ -18,6 +18,7 @@
 #define PY_FORMATTER_H_
 
 // Python.h comes first because otherwise it clobbers ceph's assert
+#include <limits.h>
 #include <Python.h>
 
 #include <stack>
--- ceph-16.2.7/src/mgr/PyModule.h.orig	2022-01-21 10:24:26.476819166 -0500
+++ ceph-16.2.7/src/mgr/PyModule.h	2022-01-21 10:24:40.182565646 -0500
@@ -13,13 +13,15 @@
 
 #pragma once
 
+#include <limits.h>
+#include "Python.h"
+
 #include <map>
 #include <memory>
 #include <string>
 #include <vector>
 #include <boost/optional.hpp>
 #include "common/ceph_mutex.h"
-#include "Python.h"
 #include "Gil.h"
 #include "mon/MgrMap.h"
 
--- ceph-16.2.7/src/mgr/PyModuleRunner.cc.orig	2022-01-21 10:20:06.971619311 -0500
+++ ceph-16.2.7/src/mgr/PyModuleRunner.cc	2022-01-21 10:20:19.815381734 -0500
@@ -13,6 +13,7 @@
 
 
 // Python.h comes first because otherwise it clobbers ceph's assert
+#include <limits.h>
 #include <Python.h>
 
 #include "PyModule.h"
--- ceph-16.2.7/src/mgr/PyOSDMap.h.orig	2022-01-21 10:23:12.642184906 -0500
+++ ceph-16.2.7/src/mgr/PyOSDMap.h	2022-01-21 10:23:57.948346865 -0500
@@ -3,6 +3,7 @@
 
 #pragma once
 
+#include <limits.h>
 #include <Python.h>
 
 #include <string>
--- ceph-16.2.7/src/mgr/PyUtil.cc.orig	2022-01-21 10:21:29.083100470 -0500
+++ ceph-16.2.7/src/mgr/PyUtil.cc	2022-01-21 10:21:39.413909379 -0500
@@ -1,6 +1,7 @@
 // -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
 // vim: ts=8 sw=2 smarttab
 
+#include <limits.h>
 #include <Python.h>
 
 #include "PyUtil.h"
--- ceph-16.2.7/src/mgr/PyUtil.h.orig	2022-01-21 10:22:43.177729918 -0500
+++ ceph-16.2.7/src/mgr/PyUtil.h	2022-01-21 10:22:56.928475568 -0500
@@ -3,10 +3,10 @@
 
 #pragma once
 
-#include <string>
-
+#include <limits.h>
 #include <Python.h>
 
+#include <string>
 #include "common/options.h"
 
 PyObject *get_python_typed_option_value(
--- ceph-16.2.7/src/mgr/StandbyPyModules.h.orig	2022-01-21 09:13:16.938691534 -0500
+++ ceph-16.2.7/src/mgr/StandbyPyModules.h	2022-01-21 10:19:22.055450138 -0500
@@ -13,11 +13,12 @@
 
 #pragma once
 
+#include <limits.h>
+#include <Python.h>
+
 #include <string>
 #include <map>
 
-#include <Python.h>
-
 #include "common/Thread.h"
 #include "common/ceph_mutex.h"
 
--- ceph-16.2.7/src/mgr/PyModule.cc.orig	2022-01-21 12:11:07.312472420 -0500
+++ ceph-16.2.7/src/mgr/PyModule.cc	2022-01-21 12:12:13.859245169 -0500
@@ -11,6 +11,7 @@
  * Foundation.  See file COPYING.
  */
 
+#include <limits.h>
 #include "BaseMgrModule.h"
 #include "BaseMgrStandbyModule.h"
 #include "PyOSDMap.h"
