--- ceph-17.1.0/cmake/modules/CheckCxxAtomic.cmake.orig	2022-03-07 15:01:49.809505648 -0500
+++ ceph-17.1.0/cmake/modules/CheckCxxAtomic.cmake	2022-03-09 12:47:31.018446185 -0500
@@ -21,6 +21,7 @@
 // We specifically test access via an otherwise unknown pointer here
 // to ensure we get the most complex case.  If this access can be
 // done without libatomic, then all accesses can be done.
+bool atomic16(std::atomic<unsigned __int128> *ptr) __attribute__ ((used));
 bool atomic16(std::atomic<unsigned __int128> *ptr)
 {
   return *ptr != 0;
