--- ceph-17.2.6/cmake/modules/BuildBoost.cmake.orig	2023-04-27 14:00:28.239524778 -0400
+++ ceph-17.2.6/cmake/modules/BuildBoost.cmake	2023-04-28 07:49:59.743342207 -0400
@@ -63,7 +63,11 @@
   else()
     list(APPEND boost_features "address-model=32")
   endif()
-  set(BOOST_CXXFLAGS "-fPIC -w") # check on arm, etc <---XXX
+  if(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64")
+    set(BOOST_CXXFLAGS "-fPIC -w -fcf-protection") # check on arm, etc <---XXX
+  else()
+    set(BOOST_CXXFLAGS "-fPIC -w") # check on arm, etc <---XXX
+  endif()
   list(APPEND boost_features "cxxflags=${BOOST_CXXFLAGS}")
 
   set(boost_with_libs)
